app/models/user.rb

```diff
@@ -82,7 +82,12 @@ def self.from_omniauth(access_token)
 
   def toggle_disabled_by_fund
     # Since toggle skips callbacks...
-    update disabled_by_fund: !disabled_by_fund
+    attrs = { disabled_by_fund: !disabled_by_fund }
+    # also, if unlocking, count that as a sign in for locking purposes
+    if disabled_by_fund?
+      attrs.merge!({ current_sign_in_at: Time.zone.now })
+    end
```

Comment:
I find this a bit hard to read - I thought we had this backwards at first! Relying on the current state of the thing we are in the middle of toggling is what we're doing, right? A quick suggestion for the explainer comment:
```suggestion
    # if we're currently locked, count this unlocking as a sign-in to avoid rapid relocking
    if disabled_by_fund?
      attrs.merge!({ current_sign_in_at: Time.zone.now })
    end
```